diff -ur ecl-0.9j.orig/src/h/object.h ecl-0.9j/src/h/object.h
--- ecl.orig/src/h/object.h	2007-12-25 23:29:03.000000000 +0100
+++ ecl/src/h/object.h	2008-01-01 18:46:48.000000000 +0100
@@ -22,8 +22,13 @@
 	Integer and boolean types (see config.h)
 */
 
+#ifndef TRUE
 #define	TRUE		1	/*  boolean true value  */
+#endif
+
+#ifndef FALSE
 #define	FALSE		0	/*  boolean false value  */
+#endif
 
 #if !defined(__cplusplus) && !defined(bool)
 typedef int bool;
