--- configure.old	2010-08-28 20:50:06.000000000 +0200
+++ configure	2010-08-29 00:07:47.000000000 +0200
@@ -160,6 +160,7 @@
 
 #*--- sqlite library (left blank for autoconfiguration) ---------------*/
 sqlitelibs=
+sqlitesrfi="sqlite"
 
 #*--- dbus library (left blank for autoconfiguration) -----------------*/
 dbuslibs=
@@ -873,6 +874,10 @@
     --hostsh=*)
       hostsh="`echo $1 | sed 's/^[^=]*=//'`";;
 
+    # [Cyp/20100828] sqlite/sqltiny have to be managed separately
+    --sqlite-backend=sqlite|--sqlite-backend=sqltiny)
+      sqlitesrfi="`echo $1 | sed 's/^[^=]*=//'`";;
+
     -*)
       if [ "$1" != "-h" -a "$1" != "-help" -a "$1" != "--help" ]; then
         echo "*** ERROR: Illegal option \"$1\""
@@ -997,6 +1002,7 @@
       echo "   --disable-gmp (see --no-gmp)" >&2;
       echo "   --customgmp=yes|no [default yes]" >&2;
       echo "   --gmpsrc=file [default \$GMP\")" >&2;
+      echo "   --sqlite-backend=sqlite|sqltiny (defaults to sqlite if available)" >&2;
       echo "" >&2;
       echo "Configuration settings:"  >&2;
       echo "   --bigloo.h[=file]" >&2;
@@ -1145,6 +1151,13 @@
     exit 1
 fi
 
+# [Cyp/20100828] Check is the sqlite backend is correct
+if [ $sqlitesrfi != "sqlite" -a $sqlitesrfi != "sqltiny" ]; then
+    echo "*** ERROR: Invalid sqlite backend."
+    echo "Correct values are: sqlite, sqltiny."
+    exit 1
+fi
+
 #*---------------------------------------------------------------------*/
 #*    Configure bootstrap                                              */
 #*---------------------------------------------------------------------*/
@@ -1804,8 +1817,10 @@
 #*---------------------------------------------------------------------*/
 # find out the C library for sqlite
 if [ "$sqlite_is_enabled " = "yes " ]; then
-  if [ "$sqlitelibs " = " " ]; then
-    sqlitelibs=`$autoconf sqlite --cflags="$cflags"` || exit 1;
+  if [ "$sqlitesrfi" = "sqlite" ]; then
+    if [ "$sqlitelibs " = " " ]; then
+       sqlitelibs=`$autoconf sqlite --cflags="$cflags"` || exit 1;
+    fi
   fi
 fi
 
@@ -2966,13 +2981,7 @@
 
   # SQLITE
   echo "SQLITELIBS=$sqlitelibs" >> $makefile_cfg
-  if [ "$sqlitelibs " = " " ]; then
-    sqlitesrfi=sqltiny;
-    echo "SQLITESRFI=sqltiny" >> $makefile_cfg
-  else
-    sqlitesrfi=sqlite;
-    echo "SQLITESRFI=sqlite" >> $makefile_cfg
-  fi
+  echo "SQLITESRFI=$sqlitesrfi" >> $makefile_cfg
 
   echo ";; automatically generated. Don't edit" > api/sqlite/src/Misc/sqlite.init
   cat api/sqlite/src/Misc/sqlite.init.in \
