diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -200,8 +200,16 @@ boot: checkgmake
          export DYLD_LIBRARY_PATH; \
          export PATH; \
          $(MAKE) -C bde boot BFLAGS="$(BFLAGS) -lib-dir $(BOOTLIBDIR) $(SHRD_BDE_OPT)" && \
-         $(MAKE) -C api boot BFLAGS="$(BFLAGS) -lib-dir $(BOOTLIBDIR)" && \
-         $(MAKE) -C bglpkg BFLAGS="$(BFLAGS) -lib-dir $(BOOTLIBDIR)")
+         $(MAKE) -C api boot BFLAGS="$(BFLAGS) -lib-dir $(BOOTLIBDIR)")
+	if [ "$(ENABLE_BGLPKG)" = "yes" ]; then \
+           PATH=$(BGLBUILDBINDIR):$(BGLBUILDLIBDIR):$$PATH; \
+           LD_LIBRARY_PATH=$(BGLBUILDLIBDIR):$$LD_LIBRARY_PATH; \
+           export LD_LIBRARY_PATH; \
+           DYLD_LIBRARY_PATH=$(BGLBUILDLIBDIR):$$DYLD_LIBRARY_PATH; \
+           export DYLD_LIBRARY_PATH; \
+           export PATH; \
+           $(MAKE) -C bglpkg BFLAGS="$(BFLAGS) -lib-dir $(BOOTLIBDIR)"; \
+         fi
 	@ echo "Boot done..."
 	@ echo "-------------------------------"
 
@@ -685,7 +696,9 @@ install-devel: install-dirs
          export LD_LIBRARY_PATH; \
          export DYLD_LIBRARY_PATH; \
 	 $(MAKE) -C bde install)
-	$(MAKE) -C bglpkg install
+	if [ "$(ENABLE_BGLPKG)" = "yes" ]; then \
+	  $(MAKE) -C bglpkg install; \
+	fi
 	$(MAKE) -C autoconf install
 
 install-libs: install-dirs
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -278,6 +278,10 @@ gcfinalizer=0
 hostsh=
 buildsh=
 
+#*--- bglpkg compilation ----------------------------------------------*/
+# valid values are "no", "yes" or "force"
+enable_bgpkg="yes"
+
 #*---------------------------------------------------------------------*/
 #*                 !!! DONT EDIT AFTER THIS LINE !!!                   */
 #*---------------------------------------------------------------------*/
@@ -847,6 +851,12 @@ while : ; do
     --gmpconfigureopt=*)
       gmpconfigureopt="`echo $1 | sed 's/^[^=]*=//'`";;
 
+    --enable-bglpkg)
+      enable_bglpkg="force";;
+
+    --disable-bglpkg)
+      enable_bglpkg="no";;
+
     --enable-srfi27)
       srfi27force="yes";;
 
@@ -1109,6 +1119,7 @@ if [ "$action" != "bigloo_config_el" ]; 
   # check apis cross dependencies
   srfi1_is_enabled=no
   packrat_is_enabled=no
+  pkgcomp_is_enabled=no
   pkglib_is_enabled=no
   sqlite_is_enabled=no
   dbus_is_enabled=no
@@ -1124,6 +1135,7 @@ if [ "$action" != "bigloo_config_el" ]; 
       srfi27) srfi27_is_enabled="yes";;
       packrat) packrat_is_enabled="yes";;
       sqlite) sqlite_is_enabled="yes";;
+      pkgcomp) pkgcomp_is_enabled="yes";;
       pkglib) pkglib_is_enabled="yes";;
       dbus) dbus_is_enabled="yes";;
       pthread) pthread_is_enabled="yes";;
@@ -1166,6 +1178,68 @@ if [ "$action" != "bigloo_config_el" ]; 
   fi
 fi
 
+# Check depends for bglpkg
+if [ $enable_bglpkg = "yes" ]; then
+    if [ $pkgcomp_is_enabled = "no" ]; then
+	echo "*** WARNING: bglpkg binary requires pkgcomp library"
+	echo "Disabling bglpkg."
+	enable_bglpkg="no"
+    fi
+    if [ $pkglib_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires pkglib library"
+	echo "Disabling bglpkg."
+	enable_bglpkg="no"
+    fi
+    if [ $sqlite_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires sqlite library"
+	echo "Disabling bglpkg."
+	enable_bglpkg="no"
+    fi
+    if [ $web_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires web library"
+	echo "Disabling bglpkg."
+	enable_bglpkg="no"
+    fi
+fi
+
+if [ $enable_bglpkg = "force" ]; then
+    if [ $pkgcomp_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires pkgcomp library"
+	echo "Either disable bglpkg, or enable pkgcomp."
+	echo 
+	echo "Enabled apis: $apis"
+	exit 1
+    fi
+    if [ $pkglib_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires pkglib library"
+	echo "Either disable bglpkg, or enable pkglib."
+	echo 
+	echo "Enabled apis: $apis"
+	exit 1
+    fi
+    if [ $sqlite_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires sqlite library"
+	echo "Either disable bglpkg, or enable sqlite."
+	echo 
+	echo "Enabled apis: $apis"
+	exit 1
+    fi
+    if [ $web_is_enabled = "no" ]; then
+	echo "*** ERROR: bglpkg binary requires web library"
+	echo "Either disable bglpkg, or enable web."
+	echo 
+	echo "Enabled apis: $apis"
+	exit 1
+    fi
+fi
+
+# Here, bglpkg has all its dependencies, setting it just to "yes"
+#  in order to have not to double test its value
+if [ $enable_bglpkg = "force" ]; then
+    enable_bglpkg="yes"
+fi
+
+
 #*---------------------------------------------------------------------*/
 #*    Configure bootstrap                                              */
 #*---------------------------------------------------------------------*/
@@ -3089,6 +3163,8 @@ if [ $action = "all" -o $action = "Makef
              -e "s|@BGLPKGNAME@|$bglpkg|g" \
        >>  $configure
   chmod a-w $configure
+  echo "ENABLE_BGLPKG=$enable_bglpkg" >> $makefile_cfg
+  echo "" >> $makefile_cfg
 
   # Does the system support strip
   echo "STRIP=$strip" >> $makefile_cfg
