diff -Naur bigloo3.1b/bdb/blib/Makefile bigloo3.1b_new/bdb/blib/Makefile
--- bigloo3.1b/bdb/blib/Makefile	2008-09-15 12:12:35.000000000 +0200
+++ bigloo3.1b_new/bdb/blib/Makefile	2009-02-06 11:28:17.000000000 +0100
@@ -36,6 +36,7 @@
 BTAGS		= $(BOOTBINDIR)/$(BTAGS_EXE)
 # Flags to build a heap
 BHEAPFLAGS	= -unsafe -q -mkaddheap -mkaddlib -L $(LIB) -heap-library bdb
+BFLAGS		:= $(BGLAGS) -lib-dir $(LIB)
 # C flags
 CFLAGS		= -I$(BGL_INC_DIR) -I. -I$(BOOTLIBDIR) $(CPICFLAGS) $(GCINCLUDE)
 # Bigloo flags
diff -Naur bigloo3.1b/bdl/src/Makefile bigloo3.1b_new/bdl/src/Makefile
--- bigloo3.1b/bdl/src/Makefile	2008-09-15 12:12:35.000000000 +0200
+++ bigloo3.1b_new/bdl/src/Makefile	2009-02-06 11:27:43.000000000 +0100
@@ -36,6 +36,7 @@
 BTAGS		= $(BOOTBINDIR)/$(BTAGS_EXE)
 # Flags to build a heap
 BHEAPFLAGS	= -unsafe -q -mkaddheap -mkaddlib -L $(LIB) -heap-library bdl
+BFLAGS		:= $(BFLAGS) -lib-dir $(LIB)
 
 #*---------------------------------------------------------------------*/
 #*    Scheme extended objects                                          */
