Only in cl-libxml2-0.3.1/foreign: cllibxml2.so
diff -ur cl-libxml2-0.3.1.orig/foreign/Makefile cl-libxml2-0.3.1/foreign/Makefile
--- cl-libxml2-0.3.1.orig/foreign/Makefile	2009-06-15 14:25:05.000000000 +0200
+++ cl-libxml2-0.3.1/foreign/Makefile	2009-06-16 14:40:17.000000000 +0200
@@ -6,16 +6,19 @@
 CC = gcc
 PREFIX = /usr
 
+LIBXML2_CFLAGS = $(shell pkg-config --cflags libxml-2.0)
+LIBXML2_LIBS = $(shell pkg-config --libs libxml-2.0)
+
 INSTALL = install
 INSOPTS = -s -m 755 -o 0 -g 0
 
 all: cllibxml2.so
 
 cllibxml2.so: cllibxml2.c
-	$(CC) -I/usr/include/libxml2 -shared -fPIC cllibxml2.c -o cllibxml2.so
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LIBXML2_CFLAGS) $(LIBXML2_LIBS) -shared -fPIC $< -o $@
 
 clean:
 	rm -f cllibxml2.so
 
 install: all
-	 $(INSTALL) $(INSOPTS) cllibxml2.so $(PREFIX)/lib
\ No newline at end of file
+	 $(INSTALL) $(INSOPTS) cllibxml2.so $(PREFIX)/lib
