diff --git a/contrib/sb-concurrency/tests/test-utils.lisp b/contrib/sb-concurrency/tests/test-utils.lisp
index bc21ebc..992b6d6 100644
--- a/contrib/sb-concurrency/tests/test-utils.lisp
+++ b/contrib/sb-concurrency/tests/test-utils.lisp
@@ -3,7 +3,7 @@
 #+sb-thread
 (progn
 
-(defparameter +timeout+ 30.0)
+(defparameter +timeout+ 120.0)
 
 (defun make-threads (n name fn)
   (loop for i from 1 to n
@@ -23,4 +23,4 @@
     (ignore-errors
       (terminate-thread thread))))
 
-) ;; #+sb-thread (progn ...
\ No newline at end of file
+) ;; #+sb-thread (progn ...
