diff --git a/tests/debug.impure.lisp b/tests/debug.impure.lisp
index 8ba4e91..d688023 100644
--- a/tests/debug.impure.lisp
+++ b/tests/debug.impure.lisp
@@ -207,13 +207,6 @@
 
 (with-test (:name (:throw :no-such-tag)
             :fails-on '(or
-                        (and :x86 :openbsd)
-                        (and :x86 :sunos)
-                        (and :x86 :darwin)
-                        (and :x86 :linux)
-                        (and :x86-64 :darwin)
-                        (and :x86-64 :linux)
-                        (and :x86-64 :openbsd)
                         (and :sparc :linux)
                         :alpha
                         :mips))
