--- scm-old/Makefile	2006-10-25 03:47:17.000000000 +0200
+++ scm/Makefile	2007-08-10 14:08:49.000000000 +0200
@@ -478,40 +478,41 @@
 	gzip -f $(infodir)hobbit.info
 
 install:	scm.1 db.so wbtab.scm rwb-isam.scm
-	test -d $(bindir) || mkdir $(bindir)
-	test -d $(mandir) || mkdir $(mandir)
-	test -d $(man1dir) || mkdir $(man1dir)
-	-cp scm scmlit $(bindir)
-	-strip $(bindir)scmlit
-	-cp scm.1 $(man1dir)
-	test -d $(libdir) || mkdir $(libdir)
-	test -d $(libscmdir) || mkdir $(libscmdir)
-	-cp $(ifiles) $(hobfiles) COPYING r4rstest.scm $(libscmdir)
-	test -f $(libscmdir)require.scm || \
-		cp requires.scm $(libscmdir)require.scm
+	test -d $(DESTDIR)$(bindir) || mkdir -p $(DESTDIR)$(bindir)
+	test -d $(DESTDIR)$(mandir) || mkdir -p $(DESTDIR)$(mandir)
+	test -d $(DESTDIR)$(man1dir) || mkdir -p $(DESTDIR)$(man1dir)
+	-cp scm scmlit $(DESTDIR)$(bindir)
+	-strip $(DESTDIR)$(bindir)scmlit
+	-cp scm.1 $(DESTDIR)$(man1dir)
+	test -d $(DESTDIR)$(libdir) || mkdir -p $(DESTDIR)$(libdir)
+	test -d $(DESTDIR)$(libscmdir) || mkdir -p $(DESTDIR)$(libscmdir)
+	-cp $(ifiles) $(hobfiles) COPYING r4rstest.scm $(DESTDIR)$(libscmdir)
+	test -f $(DESTDIR)$(libscmdir)require.scm || \
+		cp requires.scm $(DESTDIR)$(libscmdir)require.scm
 	-cp build build.scm mkimpcat.scm Iedline.scm $(SHOBJS) patchlvl.h\
-		$(xafiles) $(libscmdir)
-	-cp db.so wbtab.scm rwb-isam.scm $(libscmdir)
+		$(xafiles) $(DESTDIR)$(libscmdir)
+	if [ -f db.so ]; then cp db.so $(DESTDIR)$(libscmdir); fi
+	-cp wbtab.scm rwb-isam.scm $(DESTDIR)$(libscmdir)
 
 installlib:
-	test -d $(includedir) || mkdir $(includedir)
-	cp scm.h scmfig.h scmflags.h $(includedir)
-	test -d $(libdir) || mkdir $(libdir)
-	cp libscm.a $(libdir)libscm.a
+	test -d $(DESTDIR)$(includedir) || mkdir -p $(DESTDIR)$(includedir)
+	cp scm.h scmfig.h scmflags.h $(DESTDIR)$(includedir)
+	test -d $(DESTDIR)$(libdir) || mkdir -p $(DESTDIR)$(libdir)
+	cp libscm.a $(DESTDIR)$(libdir)libscm.a
 
 uninstall:
-	-rm $(bindir)scm $(bindir)scmlit
-	-rm $(man1dir)scm.1
-	-rm $(includedir)scm.h $(includedir)scmfig.h $(includedir)scmflags.h
-	-rm $(libdir)libscm.a
-	-(cd $(libscmdir); rm $(ifiles) $(hobfiles) COPYING r4rstest.scm)
-	-(cd $(libscmdir); rm build build.scm mkimpcat.scm \
+	-rm $(DESTDIR)$(bindir)scm $(DESTDIR)$(bindir)scmlit
+	-rm $(DESTDIR)$(man1dir)scm.1
+	-rm $(DESTDIR)$(includedir)scm.h $(DESTDIR)$(includedir)scmfig.h $(DESTDIR)$(includedir)scmflags.h
+	-rm $(DESTDIR)$(libdir)libscm.a
+	-(cd $(DESTDIR)$(libscmdir); rm $(ifiles) $(hobfiles) COPYING r4rstest.scm)
+	-(cd $(DESTDIR)$(libscmdir); rm build build.scm mkimpcat.scm \
 		$(SHOBJS) patchlvl.h $(xafiles))
-	-(cd $(libscmdir); rm db.so wbtab.scm rwb-isam.scm require.scm)
+	-(cd $(DESTDIR)$(libscmdir); rm db.so wbtab.scm rwb-isam.scm require.scm)
 
 uninstallinfo:
-	-rm $(infodir)scm.info.gz $(infodir)Xlibscm.info.gz\
-	    $(infodir)hobbit.info.gz
+	-rm $(DESTDIR)$(infodir)scm.info.gz $(DESTDIR)$(infodir)Xlibscm.info.gz\
+	    $(DESTDIR)$(infodir)hobbit.info.gz
 
 scm.doc:	scm.1
 	nroff -man $< | ul -tunknown >$@
