diff -ur lispbuilder-sdl.orig/sdl/font.lisp lispbuilder-sdl/sdl/font.lisp
--- lispbuilder-sdl.orig/sdl/font.lisp	2009-09-26 05:37:38.000000000 +0200
+++ lispbuilder-sdl/sdl/font.lisp	2010-12-17 15:21:07.531027714 +0100
@@ -205,7 +205,7 @@
   "Returns the maximum pixel descent of all glyphs of font `FONT`. 
 This can also be interpreted as the distance from the baseline to the bottom of the font. 
 It could be used when drawing an individual glyph relative to a bottom point, 
-by combining it with the glyph’s maxy metric to resolve the top of the rectangle used when 
+by combining it with the glyph's maxy metric to resolve the top of the rectangle used when 
 blitting the glyph on the screen. 
 
 ##### Parameters
@@ -242,7 +242,7 @@
   "Returns the number of faces 'sub-fonts' available in the font `FONT`. 
 This is a count of the number of specific fonts (based on size and style and other
  typographical features perhaps) contained in the font itself. It seems to be a useless
- fact to know, since it can’t be applied in any other `SDL_TTF` functions.
+ fact to know, since it can't be applied in any other `SDL_TTF` functions.
 
 ##### Parameters
 
@@ -311,4 +311,4 @@
   nil)
 
 (defmethod open-font (definition))
-(defmethod close-font (font))
\ No newline at end of file
+(defmethod close-font (font))
