diff -Naur slime.old/swank.lisp slime/swank.lisp
--- slime.old/swank.lisp	2007-11-23 09:59:12.000000000 +0100
+++ slime/swank.lisp	2007-11-23 10:01:29.000000000 +0100
@@ -2283,10 +2283,15 @@
 (defslimefun swank-require (module &optional filename)
   "Load the module MODULE."
   (unless (member (string module) *modules* :test #'string=)
-    (require module (or filename (module-filename module))))
+    (let ((*compile-print* nil)
+          (*compile-verbose* nil)
+          #+cmu (ext:*gc-verbose* nil))
+      (handler-bind ((style-warning #'muffle-warning)
+                     #+sbcl (sb-ext:compiler-note #'muffle-warning))
+        (require module (or filename (module-filename module))))))
   nil)
 
-(defvar *find-module* 'find-module
+(defvar *find-module* 'find-module-gentoo
   "Pluggable function to locate modules.
 The function receives a module name as argument and should return
 the filename of the module (or nil if the file doesn't exist).")
@@ -2319,6 +2324,11 @@
     (some (lambda (dir) (some #'probe-file (module-canditates name dir)))
           *load-path*)))
 
+(defun find-module-gentoo (module)
+  (or (let ((*load-path* (list (pathname "@CONTRIBDIR@"))))
+        (find-module module))
+      (find-module module)))
+
 
 ;;;; Macroexpansion
 
