diff -u alexandria-20071025.old/alexandria.asd alexandria-20071025/alexandria.asd
--- alexandria-20071025.old/alexandria.asd	2007-10-25 14:24:06.000000000 +0200
+++ alexandria-20071025/alexandria.asd	2007-10-26 22:25:56.000000000 +0200
@@ -2,8 +2,7 @@
   :version "0.0.0"
   :licence "Public Domain / 0-clause MIT"
   :components
-  ((:static-file "LICENCE")
-   (:static-file "tests.lisp")
+  ((:static-file "tests.lisp")
    (:file "package")
    (:file "definitions" :depends-on ("package"))
    (:file "strings" :depends-on ("package"))
Common subdirectories: alexandria-20071025.old/doc and alexandria-20071025/doc
