diff -ur cl-containers-0.9.11.orig/cl-containers.asd cl-containers-0.9.11/cl-containers.asd
--- cl-containers-0.9.11.orig/cl-containers.asd	2008-02-01 22:30:47.000000000 +0100
+++ cl-containers-0.9.11/cl-containers.asd	2008-02-01 22:46:10.000000000 +0100
@@ -76,10 +76,7 @@
 	    :depends-on ("iterators"))
      #+(or digitool openmcl)
      (:file "containers-readtable"
-	    :depends-on ("containers"))))
-   (:module "website"
-	    :components ((:module "source"
-				  :components ((:static-file "index.md"))))))
+	    :depends-on ("containers")))))
   :in-order-to ((test-op (load-op cl-containers-test)))
   :perform (test-op :after (op c)
                     (describe 
diff -ur cl-containers-0.9.11.orig/cl-containers-test.asd cl-containers-0.9.11/cl-containers-test.asd
--- cl-containers-0.9.11.orig/cl-containers-test.asd	2008-02-01 22:30:47.000000000 +0100
+++ cl-containers-0.9.11/cl-containers-test.asd	2008-02-01 22:46:19.000000000 +0100
@@ -16,8 +16,7 @@
 				      :depends-on ("package"))))
 	       (:module "tests"
 			:components 
-			((static-file "notes.text")
-			 (:file "priority-queues")
+			((:file "priority-queues")
 			 (:file "misc")
 			 (:file "trees"))))
   :depends-on (:cl-containers :lift))
