diff -u -r a/src/Makefile.in b//src/Makefile.in
--- a/src/Makefile.in	2010-03-17 20:22:06.000000000 +0200
+++ b//src/Makefile.in	2010-04-18 22:23:49.000000000 +0300
@@ -60,9 +60,6 @@
 	$(MAKE) install-@MAIN_VARIANT@
 
 plain-install:
-	if [ "$(DESTDIR)" != "" ]; then \
-          echo "cannot use plain-install with DESTDIR=$(DESTDIR)"; exit 1; \
-        fi
 	$(MAKE) plain-install-@MAIN_VARIANT@
 
 install-common-first:
diff -u -r a/src/mred/Makefile.in b//src/mred/Makefile.in
--- a/src/mred/Makefile.in	2010-03-17 20:22:01.000000000 +0200
+++ b//src/mred/Makefile.in	2010-04-19 14:50:10.000000000 +0300
@@ -280,8 +280,6 @@
 	$(MAKE) install-@WXVARIANT@-post-collects
 
 install-common:
-	cd ..; rm -f "$(DESTDIR)$(bindir)/mred@CGC_INSTALLED@"
-	cd ..; rm -f "$(DESTDIR)$(bindir)/mred@MMM_INSTALLED@"
 	cd ..; echo 'CXX=@CXX@' >> $(BUILDINFO)
 	cd ..; echo 'CXXFLAGS=@CXXFLAGS@ @PREFLAGS@ @COMPFLAGS@' >> $(BUILDINFO)
 	cd ..; echo 'MROPTIONS=@MROPTIONS@' >> $(BUILDINFO)
diff -u -r a/src/mzscheme/Makefile.in b//src/mzscheme/Makefile.in
--- a/src/mzscheme/Makefile.in	2010-03-17 20:22:06.000000000 +0200
+++ b//src/mzscheme/Makefile.in	2010-04-19 14:49:49.000000000 +0300
@@ -277,8 +277,6 @@
 # have to go up and install things from there.
 
 unix-install:	
-	cd ..; rm -f "$(DESTDIR)$(bindir)/mzscheme@CGC_INSTALLED@"
-	cd ..; rm -f "$(DESTDIR)$(bindir)/mzscheme@MMM_INSTALLED@"
 	cd ..; cp mzscheme/starter "$(DESTDIR)$(libpltdir)/starter"
 	cd ..; $(STRIP_DEBUG) "$(DESTDIR)$(libpltdir)/starter"
 	cd ..; echo 'CC=@CC@' > "$(BUILDINFO)"
