Author: Stelian Ionescu <sionescu@common-lisp.net>
Date: 2008-09-01
Status: Pending
Description: Fix DYNAMIC-EXTENT declaration.

diff -ur metatilities-0.6.15.orig/dev/utilities/sequences.lisp metatilities-0.6.15/dev/utilities/sequences.lisp
--- metatilities-0.6.15.orig/dev/utilities/sequences.lisp	2008-09-01 19:14:00.000000000 +0200
+++ metatilities-0.6.15/dev/utilities/sequences.lisp	2008-09-01 22:08:46.000000000 +0200
@@ -49,7 +49,7 @@
                     (nconc (rec (car list)) (rec (cdr list))))
                    (t               
                     (mapcan1 #'rec list)))))
-    (declare (dynamic-extent rec))
+    (declare (dynamic-extent #'rec))
     (if (atom list)
       list
       (rec list))))
