--- s-code.lisp	15 Nov 2005 16:07:57 +0100	1.99
+++ s-code.lisp	22 Feb 2007 19:14:30 +0100	
@@ -7829,7 +7829,7 @@
   (cond ((null expr-list)
          (fragl ((expr)) ((expr t)) () () () () () () :args))
         (t (cl:let ((full-expr-list
-                        (optif `(list ,expr ,@ expr-list)
+                        (optif `(,expr ,@ expr-list)
 				(cons expr (copy-list expr-list)))))
              (fragl ((full-expr-list)) ((items t))
 		    ((items t)
