diff --git a/contrib/slime-compiler-notes-tree.el b/contrib/slime-compiler-notes-tree.el
index 8b49d3f..1ed9b6e 100644
--- a/contrib/slime-compiler-notes-tree.el
+++ b/contrib/slime-compiler-notes-tree.el
@@ -180,4 +180,4 @@ This is used for labels spanning multiple lines."
     (delete-char 1)
     (goto-char start-mark)))
 
-(provide 'slime-complete-notes-tree)
\ No newline at end of file
+(provide 'slime-compiler-notes-tree)
