diff --git a/restas.asd b/restas.asd
index 075e50f..04a6ce3 100644
--- a/restas.asd
+++ b/restas.asd
@@ -15,7 +15,7 @@
   (operate 'load-op 'asdf-system-connections))
 
 (defsystem restas
-    :depends-on (#:hunchentoot #:routes #:garbage-pools #:alexandria #+swank #:closer-mop )
+    :depends-on (#:hunchentoot #:routes #:garbage-pools #:alexandria #+swank-archimag #:closer-mop )
     :components
     ((:module "src"
               :components
