diff --git a/restas.asd b/restas.asd
index 04a6ce3..34570c7 100644
--- a/restas.asd
+++ b/restas.asd
@@ -15,7 +15,7 @@
   (operate 'load-op 'asdf-system-connections))
 
 (defsystem restas
-    :depends-on (#:hunchentoot #:routes #:garbage-pools #:alexandria #+swank-archimag #:closer-mop )
+    :depends-on (#:hunchentoot #:routes #:garbage-pools #:alexandria )
     :components
     ((:module "src"
               :components
@@ -27,10 +27,6 @@
                (:file "module" :depends-on ("context"))
                (:file "route" :depends-on ("module" "render"))
                (:file "hunchentoot" :depends-on ("module" "memoize"))))
-     #+swank-archimag
-     (:module "slime" 
-              :components ((:file "restas-swank"))
-              :depends-on ("src"))
      (:module "optional"
               :components ((:file "optional"))
               :depends-on ("src"))))
