diff -ur cl-variates-0.9.0.orig/cl-variates.asd cl-variates-0.9.0/cl-variates.asd
--- cl-variates-0.9.0.orig/cl-variates.asd	2007-10-19 21:06:59.000000000 +0200
+++ cl-variates-0.9.0/cl-variates.asd	2008-01-19 01:46:59.000000000 +0100
@@ -19,17 +19,10 @@
   :licence "MIT Style License"
   :description "Portable Common Lisp Random Number Generation."
 
-  :components ((:static-file "COPYING")
-	       (:module 
-		"dev"
+  :components ((:module "dev"
 		:components ((:file "package")
 			     (:file "variates" 
-				    :depends-on ("package"))))
-               (:module 
-		"website"
-		:components ((:module 
-			      "source"
-			      :components ((:static-file "index.lml"))))))
+				    :depends-on ("package")))))
   :in-order-to ((test-op (load-op :cl-variates-test)))
   :perform (test-op :after (op c)
 		    (funcall
