diff -ur cl-closure-template-0.1.1.orig/closure-template.asd cl-closure-template-0.1.1/closure-template.asd
--- cl-closure-template-0.1.1.orig/closure-template.asd	2009-12-11 15:06:18.000000000 +0100
+++ cl-closure-template-0.1.1/closure-template.asd	2010-01-09 21:10:53.000000000 +0100
@@ -15,7 +15,7 @@
   (operate 'load-op 'asdf-system-connections))
 
 (defsystem closure-template
-  :depends-on (#:wiki-parser #:split-sequence)
+  :depends-on (#:wiki-parser #:split-sequence #:babel)
   :components ((:module "src"
                         :components ((:file "packages")
                                      (:file "expression" :depends-on ("packages"))
diff -ur cl-closure-template-0.1.1.orig/src/common-lisp-backend.lisp cl-closure-template-0.1.1/src/common-lisp-backend.lisp
--- cl-closure-template-0.1.1.orig/src/common-lisp-backend.lisp	2009-12-11 15:06:18.000000000 +0100
+++ cl-closure-template-0.1.1/src/common-lisp-backend.lisp	2010-01-09 21:11:08.000000000 +0100
@@ -33,7 +33,7 @@
               (if (or (char<= #\0 ch #\9) (char<= #\a ch #\z) (char<= #\A ch #\Z)
                       (find ch not-encode :test #'char=))
                   (write-char ch out)
-                  (iter (for octet in-vector (sb-ext:string-to-octets (string ch) :external-format :utf-8))
+                  (iter (for octet in-vector (babel:string-to-octets (string ch) :encoding :utf-8))
                           (format out "%~2,'0x" octet)))))
       str))
   
