diff -ur tinaa-0.5.5.orig/tinaa.asd tinaa-0.5.5/tinaa.asd
--- tinaa-0.5.5.orig/tinaa.asd	2007-04-29 01:51:15.000000000 +0200
+++ tinaa-0.5.5/tinaa.asd	2008-01-19 01:49:55.000000000 +0100
@@ -71,11 +71,7 @@
      (:file "epilogue"
 	    :depends-on ("tinaa"))
                                      
-     (:static-file "notes.text")
-     (:static-file "tinaa.css")))
-   (:module "website"
-	    :components ((:module "source"
-				  :components ((:static-file "index.lml"))))))
+     (:static-file "tinaa.css"))))
   
   :depends-on (:defsystem-compatibility
                :metatilities
