diff -Naur cl-rsm-filter-1.1b2.old/filter.lisp cl-rsm-filter-1.1b2/filter.lisp
--- cl-rsm-filter-1.1b2.old/filter.lisp	2003-09-24 05:14:55.000000000 +0200
+++ cl-rsm-filter-1.1b2/filter.lisp	2007-11-24 21:17:45.000000000 +0100
@@ -29,7 +29,7 @@
         (rsm.queue:enqueue el que)))))
 
 
-(declaim (ftype (function (list predicate) list) prune-tree))
+(declaim (ftype (function (list function) list) prune-tree))
 
 (defun prune-tree (tree pruner)
   "Returns a pruned version of <tree> where pruned elements satisfy the
@@ -70,7 +70,7 @@
     (rec tree (rsm.queue:create))))
 
 
-(declaim (ftype (function (predicate function) function) tree-hom))
+(declaim (ftype (function (function function) function) tree-hom))
 
 (defun tree-hom (pruner transformer)
   "Returns a function which takes a tree and returns a pruned, transformed copy.
