diff --git a/postmodern/query.lisp b/postmodern/query.lisp
index 3832fca..bca6e71 100644
--- a/postmodern/query.lisp
+++ b/postmodern/query.lisp
@@ -34,8 +34,8 @@
     (:alist symbol-alist-row-reader single-row)
     (:str-alists alist-row-reader all-rows)
     (:str-alist alist-row-reader single-row)
-    (:plists symbol-plist-row-reader nil)
-    (:plist symbol-plist-row-reader t)
+    (:plists symbol-plist-row-reader all-rows)
+    (:plist symbol-plist-row-reader single-row)
     (:column column-row-reader all-rows)
     (:single column-row-reader single-row)
     (:single! column-row-reader single-row!))
