diff -ur asdf-system-connections-0.8.3.orig/asdf-system-connections.asd asdf-system-connections-0.8.3/asdf-system-connections.asd
--- asdf-system-connections-0.8.3.orig/asdf-system-connections.asd	2007-01-20 23:43:08.000000000 +0100
+++ asdf-system-connections-0.8.3/asdf-system-connections.asd	2008-01-19 01:03:45.000000000 +0100
@@ -12,13 +12,7 @@
   :components 
   ((:module 
     "dev"
-    :components ((:file "asdf-system-connections")))
-   
-   (:module 
-    "website"
-    :components ((:module
-		  "source"
-		  :components ((:static-file "index.lml")))))))
+    :components ((:file "asdf-system-connections")))))
 
 
 
