diff -ur stumpwm.orig/command.lisp stumpwm/command.lisp
--- stumpwm.orig/command.lisp	2009-08-03 20:54:07.000000000 +0200
+++ stumpwm/command.lisp	2009-11-15 17:20:16.000000000 +0100
@@ -403,7 +403,7 @@
                    ("top-left" :top-left)
                    ("bottom-right" :bottom-right)
                    ("bottom-left" :bottom-left)))
-         (gravity (second (assoc (argument-pop-or-read input prompt values)) values :test 'string-equal)))
+         (gravity (second (assoc (argument-pop-or-read input prompt values) values :test 'string-equal))))
     (or gravity
         (throw 'error "No matching gravity."))))
 
