diff --git a/stumpwm.asd b/stumpwm.asd
index a1c095a..8a33834 100644
--- a/stumpwm.asd
+++ b/stumpwm.asd
@@ -4,13 +4,6 @@
   (:use :cl :asdf))
 (in-package :stumpwm-system)
 
-;; This is a hack for debian because it calls cmucl's clx
-;; cmucl-clx. *very* annoying. I don't actually know if debian still
-;; does this.
-#+cmu (progn
-	  (ignore-errors (require :cmucl-clx))
-	  (ignore-errors (require :clx)))
-
 (defsystem :stumpwm
   :name "StumpWM"
   :author "Shawn Betts <sabetts@vcn.bc.ca>"
@@ -53,5 +46,6 @@
 	       (:file "stumpwm")
 	       ;; keep this last so it always gets recompiled if
 	       ;; anything changes
-	       (:file "version")))
+	       (:file "version"))
+  :perform (load-op :before (op c) #+cmu (require :clx)))
 
