--- stumpwm-0.9.5.orig/wrappers.lisp	2008-10-21 23:07:48.000000000 +0200
+++ stumpwm-0.9.5/wrappers.lisp	2008-12-04 02:11:27.000000000 +0100
@@ -247,7 +247,7 @@
 
 #+ (and clisp (not (or linux freebsd)))
 (eval-when (eval load compile)
-  (let ((osname (os:uname-sysname (os:uname))))
+  (let ((osname (posix:uname-sysname (posix:uname))))
     (cond
       ((string= osname "Linux") (pushnew :linux *features*))
       ((string= osname "FreeBSD") (pushnew :freebsd *features*))
