diff -ur stumpwm.orig/floating-group.lisp stumpwm/floating-group.lisp
--- stumpwm.orig/floating-group.lisp	2009-11-15 17:28:10.000000000 +0100
+++ stumpwm/floating-group.lisp	2009-11-15 17:29:42.000000000 +0100
@@ -186,6 +186,7 @@
                    (xlib:drawable-y (window-xwin window)))))
        (multiple-value-bind (relx rely same-screen-p child state-mask)
            (xlib:query-pointer (window-parent window))
+         (declare (ignorable same-screen-p child))
          (let ((initial-width (xlib:drawable-width (slot-value window 'parent)))
                (initial-height (xlib:drawable-height (slot-value window 'parent))))
            (labels ((move-window-event-handler (&rest event-slots &key event-key &allow-other-keys)
diff -ur stumpwm.orig/wrappers.lisp stumpwm/wrappers.lisp
--- stumpwm.orig/wrappers.lisp	2009-11-15 17:28:10.000000000 +0100
+++ stumpwm/wrappers.lisp	2009-11-15 17:28:52.000000000 +0100
@@ -218,7 +218,9 @@
 (defun bytes-to-string (data)
   "Convert a list of bytes into a string."
   #+sbcl (handler-bind
-             ((sb-impl::octet-decoding-error #'(lambda (c) (invoke-restart 'use-value "?"))))
+             ((sb-impl::octet-decoding-error #'(lambda (c)
+                                                 (declare (ignore c))
+                                                 (invoke-restart 'use-value "?"))))
           (sb-ext:octets-to-string
            (make-array (length data) :element-type '(unsigned-byte 8) :initial-contents data)))
   #+clisp
@@ -240,7 +242,9 @@
 (defun utf8-to-string (octets)
   "Convert the list of octets to a string."
   #+sbcl (handler-bind
-             ((sb-impl::octet-decoding-error #'(lambda (c) (invoke-restart 'use-value "?"))))
+             ((sb-impl::octet-decoding-error #'(lambda (c)
+                                                 (declare (ignore c))
+                                                 (invoke-restart 'use-value "?"))))
            (sb-ext:octets-to-string 
             (coerce octets '(vector (unsigned-byte 8)))
             :external-format :utf-8))
Only in stumpwm: wrappers.lisp.orig
