diff -ur drakma-1.2.0.orig//util.lisp drakma-1.2.0/util.lisp
--- drakma-1.2.0.orig//util.lisp	2010-05-19 22:48:37.000000000 +0800
+++ drakma-1.2.0/util.lisp	2010-05-20 01:09:22.000000000 +0800
@@ -331,9 +331,9 @@
   "Attaches SSL to the stream HTTP-STREAM and returns the SSL stream
 \(which will not be equal to HTTP-STREAM)."
   #+:allegro
-  (socket:make-ssl-client-stream http-stream))
+  (socket:make-ssl-client-stream http-stream)
   #-:allegro
   (let ((s http-stream))
     (cl+ssl:make-ssl-client-stream
      (cl+ssl:stream-fd s)
-     :close-callback (lambda () (close s)))))
\ No newline at end of file
+     :close-callback (lambda () (close s)))))
