diff -Naur lift-1.3.6.old/lift.asd lift-1.3.6/lift.asd
--- lift-1.3.6.old/lift.asd	2007-11-23 21:26:28.000000000 +0100
+++ lift-1.3.6/lift.asd	2007-12-09 16:50:58.000000000 +0100
@@ -30,13 +30,7 @@
 			:depends-on ("port"))
 		 #+Ignore
 		 (:file "prototypes"
-			:depends-on ("lift"))))
-               
-               (:module 
-		"website"
-		:components ((:module "source"
-				      :components 
-				      ((:static-file "index.lml"))))))
+			:depends-on ("lift")))))
   
   :in-order-to ((test-op (load-op lift-test)))
   :depends-on ()
