diff -ur chipz-0.5.orig/chipz.asd chipz-0.5/chipz.asd
--- chipz-0.5.orig/chipz.asd	2008-01-01 23:17:10.000000000 +0100
+++ chipz-0.5/chipz.asd	2008-01-17 09:43:10.000000000 +0100
@@ -14,17 +14,8 @@
   :author "Nathan Froyd <froydnj@gmail.com>"
   :maintainer "Nathan Froyd <froydnj@gmail.com>"
   :description "A library for decompressing deflate, zlib, and gzip data"
-  :license "BSD style"
-  :components ((:static-file "NEWS")
-               (:static-file "LICENSE")
-               (:static-file "TODO")
-               (:file "package")
-               (:module "doc"
-                        :components
-                        ((:html-file "chipz")
-                         ;; XXX ASDF bogosity
-                         (:txt-file "chipz-doc")
-                         (:css-file "style")))
+  :license "MIT"
+  :components ((:file "package")
                (:file "types-and-tables" :depends-on ("package"))
                (:file "conditions" :depends-on ("package"))
                (:file "inflate-state" :depends-on ("types-and-tables"))
