diff -ur cmucl.orig/src/lisp/Config.linux_gencgc cmucl/src/lisp/Config.linux_gencgc
--- cmucl.orig/src/lisp/Config.linux_gencgc	2008-04-01 16:49:18.000000000 +0200
+++ cmucl/src/lisp/Config.linux_gencgc	2008-04-28 11:11:13.000000000 +0200
@@ -26,7 +26,7 @@
 CPPFLAGS = -D__NO_CTYPE -D_GNU_SOURCE -iquote. -iquote $(PATH1) $(RUNTIME)
 endif
 
-CFLAGS = -rdynamic -Wstrict-prototypes -Wall -O2 -g $(RUNTIME)
+CFLAGS = @CFLAGS@ -rdynamic -Wstrict-prototypes -Wall -O2 -g $(RUNTIME)
 ASFLAGS = -g $(GENCGC) $(LINKAGE_TABLE)
 NM = $(PATH1)/linux-nm
 UNDEFSYMPATTERN = -Xlinker -u -Xlinker &
diff -ur cmucl.orig/src/motif/server/Config.x86 cmucl/src/motif/server/Config.x86
--- cmucl.orig/src/motif/server/Config.x86	2007-06-11 16:52:55.000000000 +0200
+++ cmucl/src/motif/server/Config.x86	2008-04-28 11:12:03.000000000 +0200
@@ -1,4 +1,4 @@
-CFLAGS = -O2 -I/usr/X11R6/include -I. -I$(VPATH)
+CFLAGS = @CFLAGS@ -I/usr/include -I. -I$(VPATH)
 LDFLAGS = -L/usr/X11R6/lib
 LIBS = -lXm -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
 # This def assumes you are building in the same or parallel
