diff -ur cmucl.orig/src/code/hash-new.lisp cmucl/src/code/hash-new.lisp
--- cmucl.orig/src/code/hash-new.lisp	2008-01-22 18:55:30.000000000 +0100
+++ cmucl/src/code/hash-new.lisp	2008-05-12 10:34:49.000000000 +0200
@@ -1176,7 +1176,7 @@
      :test ',(hash-table-test table) :size ',(hash-table-size table)
      :rehash-size ',(hash-table-rehash-size table)
      :rehash-threshold ',(hash-table-rehash-threshold table)
-     :weak-p (hash-table-weak-p table))
+     :weak-p ,(hash-table-weak-p table))
    (let ((values nil))
      (declare (inline maphash))
      (maphash #'(lambda (key value)
