diff -Naur cmucl.orig/src/lisp/Config.linux_gencgc cmucl/src/lisp/Config.linux_gencgc
--- cmucl.orig/src/lisp/Config.linux_gencgc	2008-11-14 16:17:23.000000000 +0100
+++ cmucl/src/lisp/Config.linux_gencgc	2008-12-12 00:43:46.000000000 +0100
@@ -38,7 +38,7 @@
 CPPFLAGS = -D__NO_CTYPE -D_GNU_SOURCE -iquote. -iquote $(PATH1) $(RUNTIME)
 endif
 
-CFLAGS = -rdynamic -Wstrict-prototypes -Wall -O2 -g $(RUNTIME)
+CFLAGS = @CFLAGS@ -rdynamic -Wstrict-prototypes -Wall -O2 -g $(RUNTIME)
 ASFLAGS = -g $(RUNTIME)
 NM = $(PATH1)/linux-nm
 UNDEFSYMPATTERN = -Xlinker -u -Xlinker &
diff -Naur cmucl.orig/src/motif/server/Config.x86 cmucl/src/motif/server/Config.x86
--- cmucl.orig/src/motif/server/Config.x86	2007-06-11 16:52:55.000000000 +0200
+++ cmucl/src/motif/server/Config.x86	2008-12-12 00:44:52.000000000 +0100
@@ -1,5 +1,5 @@
-CFLAGS = -O2 -I/usr/X11R6/include -I. -I$(VPATH)
-LDFLAGS = -L/usr/X11R6/lib
+CFLAGS = @CFLAGS@ -I/usr/include -I. -I$(VPATH)
+LDFLAGS = -L/usr/lib
 LIBS = -lXm -lXt -lXext -lX11 -lSM -lICE -lXp -ldl -lpthread
 # This def assumes you are building in the same or parallel
 # tree to the CVS souce layout. Sites may need to customize
