diff -ur cl-graph-0.10.1.orig/cl-graph.asd cl-graph-0.10.1/cl-graph.asd
--- cl-graph-0.10.1.orig/cl-graph.asd	2008-09-29 20:14:13.000000000 +0200
+++ cl-graph-0.10.1/cl-graph.asd	2008-09-29 21:11:06.000000000 +0200
@@ -32,8 +32,10 @@
                         :depends-on ("graph"))
                  (:file "graph-matrix"
                         :depends-on ("graph"))
-                 (:file "graph-algorithms"
+                 (:file "subgraph-containing"
                         :depends-on ("graph"))
+                 (:file "graph-algorithms"
+                        :depends-on ("graph" "subgraph-containing"))
                  (:module "graphviz" :depends-on ("graph")
                           :components ((:file "graphviz-support"))))))
   :in-order-to ((test-op (load-op :cl-graph-test)))
@@ -42,6 +44,8 @@
 		      (intern (symbol-name '#:run-tests) :lift)
 		      :config :generic))
   :depends-on ((:version :metatilities-base "0.6.0")
+               (:version :moptilities "0.3.10")
+               (:version :metatilities "0.6.15")
                (:version :cl-containers "0.11.0")
                ))
 
@@ -85,14 +89,6 @@
                 ((:file "graph-metrics")))))
 
 #+asdf-system-connections
-(asdf:defsystem-connection cl-graph-and-moptilities
-  :requires (cl-graph moptilities)
-  :components ((:module
-                "dev"
-                :components
-                ((:file "subgraph-containing")))))
-
-#+asdf-system-connections
 (asdf:defsystem-connection cl-graph-and-dynamic-classes
   :requires (cl-graph dynamic-classes)
   :components ((:module
